File: 7. Reverse Integer.js
// https://leetcode.com/problems/reverse-integer/

var reverse = function (x) {
    let result = 0;

    while (x !== 0) {
        const pop = x % 10;


        if ((x / 10) | 0) {
            result = (result + pop) * 10;
        } else {
            result = result + pop;
        }

        x = (x / 10) | 0
    }


    if (result > Math.pow(2, 31) - 1 || result < Math.pow(-2, 31)) {
        return 0;
    } else {
        return result;
    }
};

console.log(reverse(1800))

File: Strings/7_Reverse_Integer.kt
// https://leetcode.com/problems/reverse-integer/

fun main () {
    print(reverse(-123))
}

fun reverse(x: Int): Int {
   var xxx: Long = Math.abs(x).toLong()

    var sign = if(x < 0) -1 else 1
    
    var res: Long = 0L
    while(xxx > 0) {
        var remainder: Long = xxx % 10
        res += remainder
        if(xxx / 10 > 0) res *= 10
        xxx /= 10
    }
    res *= sign

    if(res < Int.MIN_VALUE || res > Int.MAX_VALUE) return 0

    return res.toInt()
}